### DISCUSSION TITLE
Making subsamples from a single process and plotting them separately

### DISCUSSION THREAD

**Discussion by hephysicist** [columnflow/columnflow#413] [Q&A] [answered]
Hi all!
A short intro: I need to produce control plots and study Drell-Yan background contribution. In the case of my channel, I need to select muon and tau candidate pairs. One of the potential backgrounds for the analysis is Z->mumu, where one muon faking tau. Thus, I need to show its impact by splitting the DY dataset with a variable that describes the flavour of the generator-level particle (prompt muon or tau_h->mu).
One method I can think of is to introduce two categories (one from gen. muons that are processed as fakes and the other for tau_h -> mu). The categorisation is performed over a single dataset and has its own drawing parameters that overwrite default plotting options for the Drell-Yan sample. 
Another method is to split the dataset and reregister it at the step of creation of histograms. This method is more robust and flexible, so these histograms can also be used during the fitting procedure. 
Did anyone try to do something similar? Maybe there are more elegant ways of splitting the datasets somewhere in the task graph?

Best, 
Stepan.

**Comment by mafrahm** [comment, marked as answer]
Hi @hephysicist, I already wrote some code to create categories based on the number of gen-level particles by simply counting the number of pdgIds of particles from the hard-scatter process, maybe this is helpful for you: https://github.com/uhh-cms/hh2bbww/blob/master/hbw/selection/categories.py#L30-L90

But I think, in your case it makes more sense to define sub-processes for your dataset. At the moment, we just store the same id per dataset for each event in our `process_ids` Producer, but you could write a Producer that produces different ids based on e.g. the number of prompt electrons/muons/taus. The `process_id` column will then automatically be used to fill your different processes into different bins of the `process` axis: https://github.com/columnflow/columnflow/blob/master/columnflow/tasks/histograms.py#L221
The only other place (as far as I know) where we use the `process_id` column would be the `normalization_weight`, but on first look this should also work fine when defining multiple processes for a single dataset: https://github.com/columnflow/columnflow/blob/master/columnflow/production/normalization.py#L24

I hope these points are helpful to you. I have not yet tried to implement sub-processes, so there might be some more caveats. Maybe @riga can add more to the discussion.

Best
Mathis

**Comment by pkausw** [comment]
Since this is answered, I'll close the discussion at this point
